### ISSUE TITLE
Incorrect rollover digital numbers

### ISSUE THREAD

**Issue by woessmich** [jomjol/AI-on-the-edge-device#712]
Hi
I regularly see that on rollovers of a digital digit that the value jumps by one unit up.
It is hard to analyze but I see that the analog indicators are working reasonably well while the digital number is already interpreted as e.g 2 while it is still 1.8 or 1.9. It then jumps to 2.8 / 2.9 and if negative values are disallowed will not “self-correct”.
I also suspect that the rate limit is not working properly as with a water meter limited to 0.05 rate, and reading every 3 minutes these jumps should not happen.
I can imagine that reading the rolling numbers is incredibly hard.
Maybe reading counters in that way is just not possible.

**Comment by jomjol** [comment]
Here you find a description of the checking algorithm, where you can try to understand the behaviour: https://github.com/jomjol/AI-on-the-edge-device/wiki/Correction-Algorithm

In order to analyse the problem, the config.ini and the detailed log files of that timeframe is needed.

**Comment by woessmich** [comment]
Hi

I read the theory, but in practice it behaves differently.

This is the log of one of the jumps:
`2022-03-03T08:09:56: task_autodoFlow - next round - Round #224
2022-03-03T08:09:56: 	FlowControll.doFlow - ClassFlowMakeImage
2022-03-03T08:10:04: 	FlowControll.doFlow - ClassFlowAlignment
2022-03-03T08:10:11: 	FlowControll.doFlow - ClassFlowCNNGeneral
2022-03-03T08:10:25: 	FlowControll.doFlow - ClassFlowCNNGeneral
2022-03-03T08:10:34: 	FlowControll.doFlow - ClassFlowPostProcessing
2022-03-03T08:10:34: PostProcessing - Raw: 00264.9339 Value: 264.9339 Error: no error
2022-03-03T08:10:34: 	FlowControll.doFlow - ClassFlowMQTT
2022-03-03T08:10:34: sent publish successful in MQTTPublish, msg_id=22927, wasserzaehler/connection, connected
2022-03-03T08:10:34: sent publish successful in MQTTPublish, msg_id=5574, wasserzaehler/uptime, 40183
2022-03-03T08:10:34: sent publish successful in MQTTPublish, msg_id=34759, wasserzaehler/freeMem, 3016135
2022-03-03T08:10:34: sent publish successful in MQTTPublish, msg_id=58698, wasserzaehler/wifiRSSI, -75
2022-03-03T08:10:34: sent publish successful in MQTTPublish, msg_id=7718, wasserzaehler/main/value, 264.9339
2022-03-03T08:10:34: sent publish successful in MQTTPublish, msg_id=58442, wasserzaehler/main/error, no error
2022-03-03T08:10:34: sent publish successful in MQTTPublish, msg_id=18191, wasserzaehler/main/rate, 0.000000
2022-03-03T08:10:35: sent publish successful in MQTTPublish, msg_id=43516, wasserzaehler/main/raw, 00264.9339
2022-03-03T08:10:35: sent publish successful in MQTTPublish, msg_id=756, wasserzaehler/main/timestamp, 2022-03-03T08:10:04
2022-03-03T08:10:35: sent publish successful in MQTTPublish, msg_id=7655, wasserzaehler/main/json, {"value":264.9339,"raw":"00264.9339","error":"no error","rate":0.000000,"timestamp":"2022-03-03T08:10:04"}
2022-03-03T08:10:35: task_autodoFlow - round done
2022-03-03T08:10:35: CPU Temperature: 38.9
2022-03-03T08:12:56: task_autodoFlow - next round - Round #225
2022-03-03T08:12:56: 	FlowControll.doFlow - ClassFlowMakeImage
2022-03-03T08:13:04: 	FlowControll.doFlow - ClassFlowAlignment
2022-03-03T08:13:11: 	FlowControll.doFlow - ClassFlowCNNGeneral
2022-03-03T08:13:25: 	FlowControll.doFlow - ClassFlowCNNGeneral
2022-03-03T08:13:34: 	FlowControll.doFlow - ClassFlowPostProcessing
2022-03-03T08:13:34: 	FlowControll.doFlow - ClassFlowMQTT
2022-03-03T08:13:34: sent publish successful in MQTTPublish, msg_id=12983, wasserzaehler/connection, connected
2022-03-03T08:13:34: sent publish successful in MQTTPublish, msg_id=52144, wasserzaehler/uptime, 40363
2022-03-03T08:13:34: sent publish successful in MQTTPublish, msg_id=9433, wasserzaehler/freeMem, 3017751
2022-03-03T08:13:34: sent publish successful in MQTTPublish, msg_id=7275, wasserzaehler/wifiRSSI, -75
2022-03-03T08:13:34: sent publish successful in MQTTPublish, msg_id=1678, wasserzaehler/main/error, Rate too high - Read: 265.8339 - Pre: 264.9339
2022-03-03T08:13:34: sent publish successful in MQTTPublish, msg_id=39452, wasserzaehler/main/raw, 00264.8339
2022-03-03T08:13:34: sent publish successful in MQTTPublish, msg_id=9962, wasserzaehler/main/timestamp, 2022-03-03T08:10:04
2022-03-03T08:13:34: sent publish successful in MQTTPublish, msg_id=38754, wasserzaehler/main/json, {"value":"","raw":"00264.8339","error":"Rate too high - Read: 265.8339 - Pre: 264.9339","rate":"","timestamp":"2022-03-03T08:10:04"}
2022-03-03T08:13:35: task_autodoFlow - round done
2022-03-03T08:13:35: CPU Temperature: 53.3
2022-03-03T08:15:56: task_autodoFlow - next round - Round #226
2022-03-03T08:15:56: 	FlowControll.doFlow - ClassFlowMakeImage
2022-03-03T08:16:04: 	FlowControll.doFlow - ClassFlowAlignment
2022-03-03T08:16:11: 	FlowControll.doFlow - ClassFlowCNNGeneral
2022-03-03T08:16:25: 	FlowControll.doFlow - ClassFlowCNNGeneral
2022-03-03T08:16:34: 	FlowControll.doFlow - ClassFlowPostProcessing
2022-03-03T08:16:34: 	FlowControll.doFlow - ClassFlowMQTT
2022-03-03T08:16:34: sent publish successful in MQTTPublish, msg_id=34225, wasserzaehler/connection, connected
2022-03-03T08:16:34: sent publish successful in MQTTPublish, msg_id=19703, wasserzaehler/uptime, 40543
2022-03-03T08:16:34: sent publish successful in MQTTPublish, msg_id=50089, wasserzaehler/freeMem, 3017763
2022-03-03T08:16:34: sent publish successful in MQTTPublish, msg_id=49289, wasserzaehler/wifiRSSI, -78
2022-03-03T08:16:34: sent publish successful in MQTTPublish, msg_id=17879, wasserzaehler/main/error, Rate too high - Read: 265.8339 - Pre: 264.9339
2022-03-03T08:16:34: sent publish successful in MQTTPublish, msg_id=30132, wasserzaehler/main/raw, 00264.8339
2022-03-03T08:16:35: sent publish successful in MQTTPublish, msg_id=51729, wasserzaehler/main/timestamp, 2022-03-03T08:10:04
2022-03-03T08:16:35: sent publish successful in MQTTPublish, msg_id=35334, wasserzaehler/main/json, {"value":"","raw":"00264.8339","error":"Rate too high - Read: 265.8339 - Pre: 264.9339","rate":"","timestamp":"2022-03-03T08:10:04"}
2022-03-03T08:16:35: task_autodoFlow - round done
2022-03-03T08:16:35: CPU Temperature: 38.9
2022-03-03T08:18:56: task_autodoFlow - next round - Round #227
2022-03-03T08:18:56: 	FlowControll.doFlow - ClassFlowMakeImage
2022-03-03T08:19:04: 	FlowControll.doFlow - ClassFlowAlignment
2022-03-03T08:19:11: 	FlowControll.doFlow - ClassFlowCNNGeneral
2022-03-03T08:19:25: 	FlowControll.doFlow - ClassFlowCNNGeneral
2022-03-03T08:19:34: 	FlowControll.doFlow - ClassFlowPostProcessing
2022-03-03T08:19:34: 	FlowControll.doFlow - ClassFlowMQTT
2022-03-03T08:19:34: sent publish successful in MQTTPublish, msg_id=51071, wasserzaehler/connection, connected
2022-03-03T08:19:35: sent publish successful in MQTTPublish, msg_id=13816, wasserzaehler/uptime, 40724
2022-03-03T08:19:35: sent publish successful in MQTTPublish, msg_id=39110, wasserzaehler/freeMem, 3017763
2022-03-03T08:19:35: sent publish successful in MQTTPublish, msg_id=18046, wasserzaehler/wifiRSSI, -77
2022-03-03T08:19:35: sent publish successful in MQTTPublish, msg_id=21238, wasserzaehler/main/error, Rate too high - Read: 265.8392 - Pre: 264.9339
2022-03-03T08:19:35: sent publish successful in MQTTPublish, msg_id=36842, wasserzaehler/main/raw, 00264.8392
2022-03-03T08:19:35: sent publish successful in MQTTPublish, msg_id=2555, wasserzaehler/main/timestamp, 2022-03-03T08:10:04
2022-03-03T08:19:35: sent publish successful in MQTTPublish, msg_id=8765, wasserzaehler/main/json, {"value":"","raw":"00264.8392","error":"Rate too high - Read: 265.8392 - Pre: 264.9339","rate":"","timestamp":"2022-03-03T08:10:04"}
2022-03-03T08:19:35: task_autodoFlow - round done
2022-03-03T08:19:35: CPU Temperature: 38.9
2022-03-03T08:21:56: task_autodoFlow - next round - Round #228
2022-03-03T08:21:56: 	FlowControll.doFlow - ClassFlowMakeImage
2022-03-03T08:22:04: 	FlowControll.doFlow - ClassFlowAlignment
2022-03-03T08:22:11: 	FlowControll.doFlow - ClassFlowCNNGeneral
2022-03-03T08:22:25: 	FlowControll.doFlow - ClassFlowCNNGeneral
2022-03-03T08:22:34: 	FlowControll.doFlow - ClassFlowPostProcessing
2022-03-03T08:22:34: 	FlowControll.doFlow - ClassFlowMQTT
2022-03-03T08:22:34: sent publish successful in MQTTPublish, msg_id=15422, wasserzaehler/connection, connected
2022-03-03T08:22:34: sent publish successful in MQTTPublish, msg_id=27740, wasserzaehler/uptime, 40903
2022-03-03T08:22:34: sent publish successful in MQTTPublish, msg_id=18176, wasserzaehler/freeMem, 3017751
2022-03-03T08:22:34: sent publish successful in MQTTPublish, msg_id=18479, wasserzaehler/wifiRSSI, -77
2022-03-03T08:22:34: sent publish successful in MQTTPublish, msg_id=63517, wasserzaehler/main/error, Rate too high - Read: 265.8432 - Pre: 264.9339
2022-03-03T08:22:34: sent publish successful in MQTTPublish, msg_id=42901, wasserzaehler/main/raw, 00264.8432
2022-03-03T08:22:35: sent publish successful in MQTTPublish, msg_id=41927, wasserzaehler/main/timestamp, 2022-03-03T08:10:04
2022-03-03T08:22:35: sent publish successful in MQTTPublish, msg_id=65509, wasserzaehler/main/json, {"value":"","raw":"00264.8432","error":"Rate too high - Read: 265.8432 - Pre: 264.9339","rate":"","timestamp":"2022-03-03T08:10:04"}
2022-03-03T08:22:35: task_autodoFlow - round done
2022-03-03T08:22:35: CPU Temperature: 38.9
2022-03-03T08:24:56: task_autodoFlow - next round - Round #229
2022-03-03T08:24:56: 	FlowControll.doFlow - ClassFlowMakeImage
2022-03-03T08:25:04: 	FlowControll.doFlow - ClassFlowAlignment
2022-03-03T08:25:11: 	FlowControll.doFlow - ClassFlowCNNGeneral
2022-03-03T08:25:25: 	FlowControll.doFlow - ClassFlowCNNGeneral
2022-03-03T08:25:34: 	FlowControll.doFlow - ClassFlowPostProcessing
2022-03-03T08:25:34: 	FlowControll.doFlow - ClassFlowMQTT
2022-03-03T08:25:34: sent publish successful in MQTTPublish, msg_id=43264, wasserzaehler/connection, connected
2022-03-03T08:25:34: sent publish successful in MQTTPublish, msg_id=24619, wasserzaehler/uptime, 41083
2022-03-03T08:25:34: sent publish successful in MQTTPublish, msg_id=37615, wasserzaehler/freeMem, 3017763
2022-03-03T08:25:34: sent publish successful in MQTTPublish, msg_id=41885, wasserzaehler/wifiRSSI, -76
2022-03-03T08:25:34: sent publish successful in MQTTPublish, msg_id=3374, wasserzaehler/main/error, Rate too high - Read: 265.8432 - Pre: 264.9339
2022-03-03T08:25:35: sent publish successful in MQTTPublish, msg_id=7362, wasserzaehler/main/raw, 00264.8432
2022-03-03T08:25:35: sent publish successful in MQTTPublish, msg_id=31795, wasserzaehler/main/timestamp, 2022-03-03T08:10:04
2022-03-03T08:25:35: sent publish successful in MQTTPublish, msg_id=45459, wasserzaehler/main/json, {"value":"","raw":"00264.8432","error":"Rate too high - Read: 265.8432 - Pre: 264.9339","rate":"","timestamp":"2022-03-03T08:10:04"}
2022-03-03T08:25:35: task_autodoFlow - round done
2022-03-03T08:25:35: CPU Temperature: 38.9
2022-03-03T08:27:56: task_autodoFlow - next round - Round #230
2022-03-03T08:27:56: 	FlowControll.doFlow - ClassFlowMakeImage
2022-03-03T08:28:04: 	FlowControll.doFlow - ClassFlowAlignment
2022-03-03T08:28:11: 	FlowControll.doFlow - ClassFlowCNNGeneral
2022-03-03T08:28:25: 	FlowControll.doFlow - ClassFlowCNNGeneral
2022-03-03T08:28:34: 	FlowControll.doFlow - ClassFlowPostProcessing
2022-03-03T08:28:34: 	FlowControll.doFlow - ClassFlowMQTT
2022-03-03T08:28:34: sent publish successful in MQTTPublish, msg_id=19247, wasserzaehler/connection, connected
2022-03-03T08:28:34: sent publish successful in MQTTPublish, msg_id=61382, wasserzaehler/uptime, 41263
2022-03-03T08:28:34: sent publish successful in MQTTPublish, msg_id=35866, wasserzaehler/freeMem, 3017751
2022-03-03T08:28:35: sent publish successful in MQTTPublish, msg_id=4721, wasserzaehler/wifiRSSI, -77
2022-03-03T08:28:35: sent publish successful in MQTTPublish, msg_id=48400, wasserzaehler/main/error, Rate too high - Read: 265.8432 - Pre: 264.9339
2022-03-03T08:28:35: sent publish successful in MQTTPublish, msg_id=52170, wasserzaehler/main/raw, 00264.8432
2022-03-03T08:28:35: sent publish successful in MQTTPublish, msg_id=46717, wasserzaehler/main/timestamp, 2022-03-03T08:10:04
2022-03-03T08:28:35: sent publish successful in MQTTPublish, msg_id=14439, wasserzaehler/main/json, {"value":"","raw":"00264.8432","error":"Rate too high - Read: 265.8432 - Pre: 264.9339","rate":"","timestamp":"2022-03-03T08:10:04"}
2022-03-03T08:28:35: task_autodoFlow - round done
2022-03-03T08:28:35: CPU Temperature: 38.9
2022-03-03T08:30:56: task_autodoFlow - next round - Round #231
2022-03-03T08:30:56: 	FlowControll.doFlow - ClassFlowMakeImage
2022-03-03T08:31:04: 	FlowControll.doFlow - ClassFlowAlignment
2022-03-03T08:31:11: 	FlowControll.doFlow - ClassFlowCNNGeneral
2022-03-03T08:31:25: 	FlowControll.doFlow - ClassFlowCNNGeneral
2022-03-03T08:31:34: 	FlowControll.doFlow - ClassFlowPostProcessing
2022-03-03T08:31:34: PostProcessing - Raw: 00264.8432 Value: 265.8432 Error: no error
2022-03-03T08:31:34: 	FlowControll.doFlow - ClassFlowMQTT
2022-03-03T08:31:34: sent publish successful in MQTTPublish, msg_id=10310, wasserzaehler/connection, connected
2022-03-03T08:31:34: sent publish successful in MQTTPublish, msg_id=6758, wasserzaehler/uptime, 41443
2022-03-03T08:31:34: sent publish successful in MQTTPublish, msg_id=22059, wasserzaehler/freeMem, 3017747
2022-03-03T08:31:34: sent publish successful in MQTTPublish, msg_id=35278, wasserzaehler/wifiRSSI, -76
2022-03-03T08:31:35: sent publish successful in MQTTPublish, msg_id=64612, wasserzaehler/main/value, 265.8432
2022-03-03T08:31:35: sent publish successful in MQTTPublish, msg_id=37721, wasserzaehler/main/error, no error
2022-03-03T08:31:35: sent publish successful in MQTTPublish, msg_id=32990, wasserzaehler/main/rate, 0.043300
2022-03-03T08:31:35: sent publish successful in MQTTPublish, msg_id=35684, wasserzaehler/main/raw, 00264.8432
2022-03-03T08:31:35: sent publish successful in MQTTPublish, msg_id=56499, wasserzaehler/main/timestamp, 2022-03-03T08:31:04
2022-03-03T08:31:35: sent publish successful in MQTTPublish, msg_id=9311, wasserzaehler/main/json, {"value":265.8432,"raw":"00264.8432","error":"no error","rate":0.043300,"timestamp":"2022-03-03T08:31:04"}
2022-03-03T08:31:35: task_autodoFlow - round done
2022-03-03T08:31:35: CPU Temperature: 38.9
2022-03-03T08:33:56: task_autodoFlow - next round - Round #232
2022-03-03T08:33:56: 	FlowControll.doFlow - ClassFlowMakeImage
2022-03-03T08:34:04: 	FlowControll.doFlow - ClassFlowAlignment
2022-03-03T08:34:11: 	FlowControll.doFlow - ClassFlowCNNGeneral
2022-03-03T08:34:25: 	FlowControll.doFlow - ClassFlowCNNGeneral
2022-03-03T08:34:34: 	FlowControll.doFlow - ClassFlowPostProcessing
2022-03-03T08:34:34: PostProcessing - Raw: 00264.8432 Value: 265.8432 Error: no error
2022-03-03T08:34:34: 	FlowControll.doFlow - ClassFlowMQTT
2022-03-03T08:34:34: sent publish successful in MQTTPublish, msg_id=31528, wasserzaehler/connection, connected
2022-03-03T08:34:35: sent publish successful in MQTTPublish, msg_id=38336, wasserzaehler/uptime, 41624
2022-03-03T08:34:35: sent publish successful in MQTTPublish, msg_id=54887, wasserzaehler/freeMem, 3017751
2022-03-03T08:34:35: sent publish successful in MQTTPublish, msg_id=44059, wasserzaehler/wifiRSSI, -77
2022-03-03T08:34:35: sent publish successful in MQTTPublish, msg_id=9845, wasserzaehler/main/value, 265.8432
2022-03-03T08:34:35: sent publish successful in MQTTPublish, msg_id=47000, wasserzaehler/main/error, no error
2022-03-03T08:34:35: sent publish successful in MQTTPublish, msg_id=52521, wasserzaehler/main/rate, 0.000000
2022-03-03T08:34:35: sent publish successful in MQTTPublish, msg_id=24771, wasserzaehler/main/raw, 00264.8432
2022-03-03T08:34:35: sent publish successful in MQTTPublish, msg_id=38110, wasserzaehler/main/timestamp, 2022-03-03T08:34:04
2022-03-03T08:34:35: sent publish successful in MQTTPublish, msg_id=57101, wasserzaehler/main/json, {"value":265.8432,"raw":"00264.8432","error":"no error","rate":0.000000,"timestamp":"2022-03-03T08:34:04"}
2022-03-03T08:34:35: task_autodoFlow - round done
2022-03-03T08:34:35: CPU Temperature: 53.3
2022-03-03T08:36:56: task_autodoFlow - next round - Round #233
2022-03-03T08:36:56: 	FlowControll.doFlow - ClassFlowMakeImage
2022-03-03T08:37:04: 	FlowControll.doFlow - ClassFlowAlignment
2022-03-03T08:37:11: 	FlowControll.doFlow - ClassFlowCNNGeneral
2022-03-03T08:37:25: 	FlowControll.doFlow - ClassFlowCNNGeneral
2022-03-03T08:37:34: 	FlowControll.doFlow - ClassFlowPostProcessing
2022-03-03T08:37:34: PostProcessing - Raw: 00264.8432 Value: 265.8432 Error: no error
2022-03-03T08:37:34: 	FlowControll.doFlow - ClassFlowMQTT
2022-03-03T08:37:34: sent publish successful in MQTTPublish, msg_id=39735, wasserzaehler/connection, connected
2022-03-03T08:37:34: sent publish successful in MQTTPublish, msg_id=5101, wasserzaehler/uptime, 41803
2022-03-03T08:37:35: sent publish successful in MQTTPublish, msg_id=23624, wasserzaehler/freeMem, 3017747
2022-03-03T08:37:35: sent publish successful in MQTTPublish, msg_id=44633, wasserzaehler/wifiRSSI, -74
2022-03-03T08:37:35: sent publish successful in MQTTPublish, msg_id=2099, wasserzaehler/main/value, 265.8432
2022-03-03T08:37:35: sent publish successful in MQTTPublish, msg_id=14877, wasserzaehler/main/error, no error
2022-03-03T08:37:35: sent publish successful in MQTTPublish, msg_id=33376, wasserzaehler/main/rate, 0.000000
2022-03-03T08:37:35: sent publish successful in MQTTPublish, msg_id=62200, wasserzaehler/main/raw, 00264.8432
2022-03-03T08:37:35: sent publish successful in MQTTPublish, msg_id=24688, wasserzaehler/main/timestamp, 2022-03-03T08:37:04
2022-03-03T08:37:35: sent publish successful in MQTTPublish, msg_id=36581, wasserzaehler/main/json, {"value":265.8432,"raw":"00264.8432","error":"no error","rate":0.000000,"timestamp":"2022-03-03T08:37:04"}
2022-03-03T08:37:35: task_autodoFlow - round done
2022-03-03T08:37:35: CPU Temperature: 39.4`

And the config.ini:
[MakeImage]
;LogImageLocation = /log/source
WaitBeforeTakingPicture = 5
LogfileRetentionInDays = 2
Brightness = 0
Contrast = 0
Saturation = 0
LEDIntensity = 15
ImageQuality = 10
ImageSize = VGA
FixedExposure = false

[Alignment]
InitialRotate = 90.2
InitialMirror = false
SearchFieldX = 20
SearchFieldY = 20
AlignmentAlgo = highAccurity
FlipImageSize = true
/config/ref0.jpg 437 285
/config/ref1.jpg 32 416

[Digits]
Model = /config/dig1330s1q.tflite
;LogImageLocation = /log/digit
LogfileRetentionInDays = 3
ModelInputSize = 20 32
main.dig1 97 237 34 54
main.dig2 144 233 34 54
main.dig3 193 233 34 54
main.dig4 243 233 32 53
main.dig5 290 233 32 53

[Analog]
Model = /config/ana0910s3_longq.tflite
;LogImageLocation = /log/analog
LogfileRetentionInDays = 2
ModelInputSize = 32 32
main.ana1 336 299 102 102
main.ana2 300 415 102 102
main.ana3 189 462 102 102
main.ana4 76 418 102 102

[PostProcessing]
main.DecimalShift = 0
PreValueUse = true
PreValueAgeStartup = 720
AllowNegativeRates = true
main.MaxRateValue = 0.05
main.MaxRateType = RateChange
;main.ExtendedResolution = false
;main.IgnoreLeadingNaN = false
ErrorMessage = true
CheckDigitIncreaseConsistency = true

[MQTT]
Uri = mqtt://192.168.1.10:1883
MainTopic = wasserzaehler
ClientID = wasser
user = mosquitto
password = ******

;[GPIO]
;IO0 = input disabled 10 false false 
;IO1 = input disabled 10 false false 
;IO3 = input disabled 10 false false 
;IO4 = built-in-led disabled 10 false false 
;IO12 = input-pullup disabled 10 false false 
;IO13 = input-pullup disabled 10 false false 
LEDType = WS2812
LEDNumbers = 2
LEDColor = 150 150 150

[AutoTimer]
AutoStart = true
Intervall = 3

[Debug]
Logfile = true
LogfileRetentionInDays = 3

[System]
TimeZone = CET-1CEST,M3.5.0,M10.5.0/3
;TimeServer = 
;AutoAdjustSummertime = false
;Hostname = undefined
SetupMode = false

**Comment by friedpa** [comment]
Sorry, your log is not readable. Please format in a way that the timestamp is always left.....
This behaviour was descript already many times, you MUST set your MaxRate right.....

**Comment by woessmich** [comment]
This is more a problem of code formatting here than the log itself. I now formatted it as pure text so it won't get wrapped around.

"right" means what?
What is supposedly wrong with a maximum change rate of 0.05 which is supposed to be 50 Liters per minute for a water counter?
The system is correctly rejecting a rate of >265 for a while while it is reading 265.8xxx for a while and then at 08:31 it is deciding that the raw value of 264.8432 is actually 265.8432.
{"value":265.8432,"raw":"00264.8432","error":"no error","rate":0.043300,"timestamp":"2022-03-03T08:31:04"}

**Comment by jomjol** [comment]
@woessmich : checking your log file, the algo is exactly doing as expected and set in the config:
```
2022-03-03T08:10:35:   sent publish successful in MQTTPublish, msg_id=7655, wasserzaehler/main/json,   {"value":264.9339,"raw":"00264.9339","error":"no   error","rate":0.000000,"timestamp":"2022-03-03T08:10:04"}
--
2022-03-03T08:13:34: sent publish   successful in MQTTPublish, msg_id=38754, wasserzaehler/main/json,   {"value":"","raw":"00264.8339","error":"Rate   too high - Read: 265.8339 - Pre: 264.9339","rate":"","timestamp":"2022-03-03T08:10:04"}
2022-03-03T08:16:35: sent publish   successful in MQTTPublish, msg_id=35334, wasserzaehler/main/json,   {"value":"","raw":"00264.8339","error":"Rate   too high - Read: 265.8339 - Pre: 264.9339","rate":"","timestamp":"2022-03-03T08:10:04"}
2022-03-03T08:19:35: sent publish   successful in MQTTPublish, msg_id=8765, wasserzaehler/main/json,   {"value":"","raw":"00264.8392","error":"Rate   too high - Read: 265.8392 - Pre: 264.9339","rate":"","timestamp":"2022-03-03T08:10:04"}
2022-03-03T08:22:35: sent publish   successful in MQTTPublish, msg_id=65509, wasserzaehler/main/json,   {"value":"","raw":"00264.8432","error":"Rate   too high - Read: 265.8432 - Pre: 264.9339","rate":"","timestamp":"2022-03-03T08:10:04"}
2022-03-03T08:25:35: sent publish   successful in MQTTPublish, msg_id=45459, wasserzaehler/main/json,   {"value":"","raw":"00264.8432","error":"Rate   too high - Read: 265.8432 - Pre: 264.9339","rate":"","timestamp":"2022-03-03T08:10:04"}
2022-03-03T08:28:35: sent publish   successful in MQTTPublish, msg_id=14439, wasserzaehler/main/json,   {"value":"","raw":"00264.8432","error":"Rate   too high - Read: 265.8432 - Pre: 264.9339","rate":"","timestamp":"2022-03-03T08:10:04"}
2022-03-03T08:31:35: sent publish   successful in MQTTPublish, msg_id=9311, wasserzaehler/main/json,   {"value":265.8432,"raw":"00264.8432","error":"no   error","rate":0.043300,"timestamp":"2022-03-03T08:31:04"}
2022-03-03T08:34:35: sent publish   successful in MQTTPublish, msg_id=57101, wasserzaehler/main/json,   {"value":265.8432,"raw":"00264.8432","error":"no   error","rate":0.000000,"timestamp":"2022-03-03T08:34:04"}
2022-03-03T08:37:35: sent publish   successful in MQTTPublish, msg_id=36581, wasserzaehler/main/json,   {"value":265.8432,"raw":"00264.8432","error":"no   error","rate":0.000000,"timestamp":"2022-03-03T08:37:04"}
```
8:10 - last correct value: 264.**9**339
after that a "zero crossing" due to 264.**8339 result in a correction to 265.8339, which was rejected due to the too right rate.
You set a `ChangeRate` based limit for the max allowed rate. After about 20 minutes (8:31) the max rate was 20minutes * 0.05 m³/minute = 1 m³ and there there was no "Rate too high" error anymore and the value was accepted.
You can avoid this by an absolut setting of the rate limit.

Regarding your question with the `CheckDigitIncreaseConsistency`. This is only applied to the digital values, but it uses the last analog as an input.

**Comment by woessmich** [comment]
Thank you for the support!
With absolute change rate instead of relative it now seems to work.

I have to admit that configuration of this specific detail is very difficult as it is hard to test and to understand. The fact that there might be no readings for a while during rollover and the relative change adding up makes it less simple to configure compared to just setting the maximum expected value change per minute. This is what the explanations suggest.

**Comment by philsson** [comment]
I have this problem too and would appreciate some guidance on how to solve it. Today my reading on the last digit was between 2 and 3 and being read as 0, hence "Error - Negative rate". After it flipped to 3 the rate limiter kicks in and I have to manually reset the pre value. This happens now and again to me.

My config below:

```txt
[MakeImage]
;LogImageLocation = /log/source
WaitBeforeTakingPicture = 5
;LogfileRetentionInDays = 15
Brightness = -2
Contrast = 2
Saturation = 1
LEDIntensity = 4
ImageQuality = 5
ImageSize = VGA
FixedExposure = false

[Alignment]
InitialRotate = -7
InitialMirror = false
SearchFieldX = 20
SearchFieldY = 20
AlignmentAlgo = default
FlipImageSize = false
/config/ref0.jpg 129 119
/config/ref1.jpg 495 134

[Digits]
Model = /config/dig1330s1q.tflite
;CNNGoodThreshold = 
;LogImageLocation = /log/digit
;LogfileRetentionInDays = 3
main.dig1 239 116 22 39
main.dig2 274 116 22 39
main.dig3 308 116 22 39
main.dig4 342 116 22 39
main.dig5 376 116 22 39

[Analog]
Model = /config/ana0910s3_longq.tflite
;LogImageLocation = /log/analog
;LogfileRetentionInDays = 3
main.ana1 453 181 83 82
main.ana2 414 276 83 84
main.ana3 318 315 82 82
main.ana4 218 274 82 82

[PostProcessing]
main.DecimalShift = 0
PreValueUse = true
PreValueAgeStartup = 720
AllowNegativeRates = false
main.MaxRateValue = 0.1
main.MaxRateType = AbsoluteChange
;main.ExtendedResolution = false
;main.IgnoreLeadingNaN = false
ErrorMessage = true
CheckDigitIncreaseConsistency = false

[MQTT]
Uri = mqtt://proxmox.lan:1883
MainTopic = watermeter
ClientID = wasser
user = mqttjoh
password = dJSC496WW
;SetRetainFlag = 

;[InfluxDB]
;Uri = 
;Database = 
;Measurement = 
;user = 
;password = 

;[GPIO]
;IO0 = input disabled 10 false false 
;IO1 = input disabled 10 false false 
;IO3 = input disabled 10 false false 
;IO4 = built-in-led disabled 10 false false 
;IO12 = input-pullup disabled 10 false false 
;IO13 = input-pullup disabled 10 false false 
LEDType = WS2812
LEDNumbers = 2
LEDColor = 50 50 50

[AutoTimer]
AutoStart = true
Intervall = 4.85

[Debug]
Logfile = true
LogfileRetentionInDays = 3

[System]
TimeZone = CET-1CEST,M3.5.0,M10.5.0/3
;TimeServer = undefined
;AutoAdjustSummertime = false
;Hostname = undefined
SetupMode = false
```

**Comment by jomjol** [comment]
at which position is your last digit? max rate value is set to 0.1. might this cause the problem?

**Comment by philsson** [comment]
> at which position is your last digit? max rate value is set to 0.1. might this cause the problem?

Excuse this late reply, but I think this is about to happen again. The last digital digit is showing `5` when analog ones have not yet rolled over. Last accurate reading (previous value) was 994.9750 and the current raw is 995.9956. I don't know the unit of the rate limiter and if the time is calculated from the time of "previous value", but if it is then the rate may be quite low once it reads correctly again. Even nicer would be if the software could figure out that we are still at 994 and correct for it.
![image](https://user-images.githubusercontent.com/7071234/179969474-912cd9f6-a897-4382-9ea1-73303d7e7f6c.png)

**Comment by haverland** [comment]
Could you send your Version you use and the names of the models?

**Comment by haverland** [comment]
OK. I can reproduce it. Digital models (not Hyprid or 100) predicts a x.0. The new ZeigerEval-Version for the newer models do not pay attention to that.
@jomjol: See pull request on rolling

**Comment by philsson** [comment]
> Could you send your Version you use and the names of the models?

I'm running `master - v10.5.2 - 2022-02-22`. Where can I find what models I'm using? I may not have updated the content on the SD card since I installed this a while back.

**Comment by philsson** [comment]
> You can update to 10.6.0. I had a mistake in understood your issue. The new version should work fine.

Nice :)
I've now tried updating but the info says I'm still on `10.5.2`. I have downloaded and uploaded (and flashed) twice now and even tried checking the info in incognito mode on my browser. Could it be that the tags are not updated properly in the latest release? This is how the download page looks
![image](https://user-images.githubusercontent.com/7071234/180067896-4839c033-0501-45f3-9897-c87b4d55146a.png)


EDIT: I've managed to get the info right after building locally from source
![image](https://user-images.githubusercontent.com/7071234/180067977-935dc1a6-1726-461c-9d40-4f97ba54c243.png)
